File: java/674_Longest_Continuous_Increasing_Subsequence.java
/*
https://leetcode.com/problems/longest-continuous-increasing-subsequence/
Given an unsorted array of integers, find the length of longest continuous increasing subsequence (subarray).

Iterate over array. If num is greater than previous, increment current length. Else update longest length and reset
current sequence.
Time - O(n)
Space - O(1)
*/

class Solution {
    public int findLengthOfLCIS(int[] nums) {

        if (nums.length == 0)
            return 0;

        int current = 1;
        int longest = 1;
        for (int i = 1; i < nums.length; ++i) {
            if (nums[i] > nums[i - 1])
                ++current;
            else {
                longest = Math.max(longest, current);
                current = 1;
            }
        }

        return Math.max(longest, current);
    }
}

File: python_1_to_1000/674_Longest_Continuous_Increasing_Subsequence.py
_author_ = 'jake'
_project_ = 'leetcode'

# https://leetcode.com/problems/longest-continuous-increasing-subsequence/
# Given an unsorted array of integers, find the length of longest continuous increasing subsequence (subarray).

# Dynamic programming. Find longest CIS ending at each num. If non-increasing, reset current sequence length.
# Time - O(n)
# Space - O(1)

class Solution(object):
    def findLengthOfLCIS(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        longest, current = 0, 0

        for i, num in enumerate(nums):

            if i == 0 or num <= nums[i - 1]:
                current = 0

            current += 1
            longest = max(longest, current)

        return longest